src/libs/actions/Report.js

```diff
@@ -1250,32 +1283,69 @@ function editReportComment(reportID, originalReportAction, textForNewComment) {
         return;
     }
 
-    // Optimistically update the report action with the new message
+    // Optimistically update the reportAction with the new message
     const sequenceNumber = originalReportAction.sequenceNumber;
-    const newReportAction = {...originalReportAction};
-    const actionToMerge = {};
-    newReportAction.message[0].isEdited = true;
-    newReportAction.message[0].html = htmlForNewComment;
-    newReportAction.message[0].text = parser.htmlToText(htmlForNewComment);
-    actionToMerge[sequenceNumber] = newReportAction;
-    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, actionToMerge);
-
-    // Persist the updated report comment
-    DeprecatedAPI.Report_EditComment({
+    const optimisticReportActions = {
+        [sequenceNumber]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+            message: [{
+                isEdited: true,
+                html: htmlForNewComment,
+                text: textForNewComment,
+            }],
+        },
+    };
+
+    // Optimistically update the report's last message if the user edited it
+    const highestSequenceNumber = getMaxSequenceNumber(reportID);
+    const optimisticReport = {};
+    if (sequenceNumber === highestSequenceNumber) {
+        optimisticReport.lastMessageText = ReportUtils.formatReportLastMessageText(textForNewComment);
+    }
+
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: optimisticReportActions,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: optimisticReport,
```

Comment:
NAB, if this can be an empty object better to not do the Onyx update at all and removed from the `optimisticData` array.